salvo/src/lib/test_docker_image.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+"""
+Test Docker interactions
+"""
+
+import os
+import re
+import site
+import pytest
+
+site.addsitedir("src")
+
+from lib.docker_image import DockerImage
+
+
+def test_pull_image():
+  """Test retrieving an image"""
+
+  if not os.path.exists("/var/run/docker.sock"):
+    pytest.skip("Skipping docker test since no socket is available")
+
+  docker_image = DockerImage()
+  container = docker_image.pull_image("oschaaf/benchmark-dev:latest")
+  assert container is not None
+
+
+def test_run_image():
+  """Test executing a command in an image"""
+
+  if not os.path.exists("/var/run/docker.sock"):
+    pytest.skip("Skipping docker test since no socket is available")
+
+  env = ['key1=val1', 'key2=val2']
+  cmd = ['uname', '-r']
+  image_name = 'oschaaf/benchmark-dev:latest'
```

Comment:
How come Otto's image is hardcoded?

Reply:
It was available in dockerhub and I used it to validate that the docker module could actually pull an image and execute a command.  The actual image here is moot and could be any valid docker image.